src/sdk/namespace_fs.js

```diff
@@ -1332,6 +1346,29 @@ class NamespaceFS {
         };
     }
 
+    async _get_version_path(params, fs_context) {
+        if (this.versioning === 'ENABLED' || this.versioning === 'SUSPENDED') {
+            const file_path = this._get_file_path(params);
+            const stat = await nb_native().fs.stat(fs_context, file_path);
+            if (isDirectory(stat)) throw Object.assign(new Error('NoSuchKey'), { code: 'ENOENT' });
+            const xattr_version_id = stat.xattr[XATTR_VERSION_ID];
+            if (!xattr_version_id || !params.version_id || xattr_version_id === params.version_id) {
```

Comment:
A few comments about that if - 
1. no need to stat if !params.version_id, you can move this check to the condition in the first line, and the versioning mode check is not needed because the user can ask for a version id even if the versioning is still disabled and we should fail since the version doesn't exist, but we shouldn't return the latest version. 
2. The first condition here !xattr_version_id is incorrect, for example, if a user asked for a specific version but it doesn't exist and there is only latest version of that object, the latest version will return but it's not what the user asked, we should fail with no such object in this case.
3. The last condition - we should fail with no such object here also if we can't find the version in the latest version path or under .versions/ so we need to add stat check for the version path as well and fail if we couldn't find this version

Reply:
I updated the patch to align with above comments (may not be complete, Please take look)